### ISSUE TITLE
Docker image 0.4 is broken

### ISSUE THREAD

**Issue by sofixa** [hashicorp/consul-terraform-sync#474]
### Describe the bug

The Docker image is broken due to Alpine musl/wherever the binary was compiled libc incompatibilities.

### Versions
**Consul Terraform Sync**
```
hashicorp/consul-terraform-sync:0.4.0
```

### Expected Behavior

Running the docker image with:
```
docker run -it hashicorp/consul-terraform-sync:0.4.0
```
should work.

### Actual Behavior
```
/bin/docker-entrypoint.sh: exec: line 35: /bin/consul-terraform-sync: not found
```

### Steps to Reproduce
1. `docker run -it hashicorp/consul-terraform-sync:0.4.0`
2. See error

### Additional Context

Running the image with `-u root` and then `apk update && apk install libc6-compat` fixes the issue. A more appropriate fix would be to rebuild the binary in an alpine container with the golang toolchain, and pass it to an empty alpine container. `CGO_ENABLED=0` might also do the trick.

Adrian

PS: Oh and it'd be nice to have a test for this!

**Comment by findkim** [comment]
Thank you for reporting this @sofixa. We're sorry about the regression in usability here. We made a large haul to transition to a new release process (our project being the first adopter). Hopefully this is the last of the kinks to work through!

We'll look into disabling cgo first, and look into a plan for the 0.4.0 docker images. We have 0.4.1 planned to release before the end of the month.

**Comment by findkim** [comment]
Reproduced it with the 0.4.0 and 0.4.0-beta1 images

```
docker run --name cts-0.4.0 \
  -v $(pwd):/consul-terraform-sync/config \
  -e "CONSUL_HTTP_ADDR=host.docker.internal:8500" \
  --rm -it --entrypoint sh \
  hashicorp/consul-terraform-sync
/consul-terraform-sync $ whoami
consul-terraform-sync
/consul-terraform-sync $ ls -al /bin/consul-terraform-sync 
-rwxr-xr-x    1 root     root      35310693 Oct 13 19:08 /bin/consul-terraform-sync
/consul-terraform-sync $ /bin/consul-terraform-sync -version
sh: /bin/consul-terraform-sync: not found
```

Added `-u root` to the above command to run `apk update && apk add libc6-compat` and install the missing C links based on your suggestion and it resolves it. 🙂 
```
/consul-terraform-sync # whoami
root
/consul-terraform-sync # /bin/consul-terraform-sync -version
sh: /bin/consul-terraform-sync: not found
/consul-terraform-sync # ls -al /bin/consul-terraform-sync 
-rwxr-xr-x    1 root     root      35310693 Oct 13 19:08 /bin/consul-terraform-sync
/consul-terraform-sync # apk update && apk add libc6-compat
fetch https://dl-cdn.alpinelinux.org/alpine/v3.13/main/x86_64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/v3.13/community/x86_64/APKINDEX.tar.gz
v3.13.6-55-gc3411e25b0 [https://dl-cdn.alpinelinux.org/alpine/v3.13/main]
v3.13.6-54-g9a2717089a [https://dl-cdn.alpinelinux.org/alpine/v3.13/community]
OK: 13891 distinct packages available
(1/1) Installing libc6-compat (1.2.2-r1)
OK: 19 MiB in 27 packages
/consul-terraform-sync # consul-terraform-sync -version
consul-terraform-sync 0.4.0
Compatible with Terraform >= 0.13.0, < 1.1.0
```

**Comment by findkim** [comment]
Hi @sofixa. We kept the 0.4.0 release as-is, and instead released 0.4.1 today with the fix for the docker image. Our other changes will continue on schedule for later this month.
